app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawer.kt

```diff
@@ -0,0 +1,237 @@
+            event.layoutParams = lp
+
+            updateEventView(event, lecture)
+        }
+    }
+
+    fun setLectureBackground(event: Lecture, eventView: View) {
+        val context = eventView.context
+        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
+        val defaultValue = resources.getBoolean(R.bool.preferences_alternative_highlight_enabled_default_value)
+        val alternativeHighlightingIsEnabled = prefs.getBoolean(
+                BundleKeys.PREFS_ALTERNATIVE_HIGHLIGHT, defaultValue)
+        val eventIsFavored = event.highlight
+        @ColorRes val backgroundColorResId: Int
+        backgroundColorResId = if (eventIsFavored) {
+            val colorResId = trackNameBackgroundColorHighlightPairs[event.track]
+            colorResId ?: R.color.event_border_highlight
+        } else {
+            val colorResId = trackNameBackgroundColorDefaultPairs[event.track]
+            colorResId ?: R.color.event_border_default
+        }
+        @ColorInt val backgroundColor = ContextCompat.getColor(context, backgroundColorResId)
+        val eventDrawable: EventDrawable
+        eventDrawable = if (eventIsFavored && alternativeHighlightingIsEnabled) {
+            EventDrawable(
+                    backgroundColor,
+                    eventDrawableCornerRadius.toFloat(),
+                    eventDrawableRippleColor,
+                    eventDrawableStrokeColor,
+                    eventDrawableStrokeWidth.toFloat())
+        } else {
+            EventDrawable(
+                    backgroundColor,
+                    eventDrawableCornerRadius.toFloat(),
+                    eventDrawableRippleColor)
+        }
+        eventDrawable.setLayerInset(EventDrawable.BACKGROUND_LAYER_INDEX,
+                eventDrawableInsetLeft,
+                eventDrawableInsetTop,
+                eventDrawableInsetRight,
+                0)
+        eventDrawable.setLayerInset(EventDrawable.STROKE_LAYER_INDEX,
+                eventDrawableInsetLeft,
+                eventDrawableInsetTop,
+                eventDrawableInsetRight,
+                0)
+        eventView.setBackgroundDrawable(eventDrawable)
+        val padding = eventPadding
+        eventView.setPadding(padding, padding, padding, padding)
+    }
+
+    fun setLectureTextColor(lecture: Lecture, view: View) {
+        val title = view.findViewById<TextView>(R.id.event_title)
+        val subtitle = view.findViewById<TextView>(R.id.event_subtitle)
+        val speakers = view.findViewById<TextView>(R.id.event_speakers)
+        val colorResId = if (lecture.highlight) R.color.event_title_highlight else R.color.event_title
+        val textColor = ContextCompat.getColor(view.context, colorResId)
+        title.setTextColor(textColor)
+        subtitle.setTextColor(textColor)
+        speakers.setTextColor(textColor)
+    }
+
+    companion object {
+        const val LOG_TAG = "LectureViewDrawer"
+        const val DIVISOR = 5
```

Comment:
I don't know why it is 5 actually. standardHeight is 13*deviceScale. Neither do I know why 13. Since this are both magic numbers I can't tell you any better name than that, unfortunately :(

Reply:
I can't answer this myself out of the box. I just know that there is a minimum height for very short (time-wise) lectures. It might be that the same `5` is used in other parts of the code, too. If you cannot find it out then we will hopefully clarify this in the future. - Feel free to close this comment if you can't find an answer.